scio-parquet/src/main/scala/com/spotify/scio/parquet/tensorflow/ParquetExampleIO.scala

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright 2020 Spotify AB.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.spotify.scio.parquet.tensorflow
+
+import java.lang.{Boolean => JBoolean}
+
+import com.spotify.scio.ScioContext
+import com.spotify.scio.io.{ScioIO, Tap, TapOf}
+import com.spotify.scio.parquet.{BeamInputFile, GcsConnectorUtil}
+import com.spotify.scio.util.ScioUtil
+import com.spotify.scio.values.SCollection
+import me.lyh.parquet.tensorflow.{
+  ExampleParquetInputFormat,
+  ExampleParquetReader,
+  ExampleReadSupport,
+  Schema
+}
+import org.apache.beam.sdk.io.{
+  DefaultFilenamePolicy,
+  DynamicFileDestinations,
+  FileBasedSink,
+  FileSystems,
+  WriteFiles
+}
+import org.apache.beam.sdk.io.hadoop.format.HadoopFormatIO
+import org.apache.beam.sdk.options.ValueProvider.StaticValueProvider
+import org.apache.beam.sdk.transforms.SimpleFunction
+import org.apache.hadoop.mapreduce.Job
+import org.apache.parquet.filter2.predicate.FilterPredicate
+import org.apache.parquet.hadoop.ParquetInputFormat
+import org.apache.parquet.hadoop.metadata.CompressionCodecName
+import org.tensorflow.example.Example
+
+import scala.collection.JavaConverters._
+
+final case class ParquetExampleIO(path: String) extends ScioIO[Example] {
+  override type ReadP = ParquetExampleIO.ReadParam
+  override type WriteP = ParquetExampleIO.WriteParam
+  override val tapT = TapOf[Example]
+
+  override protected def read(sc: ScioContext, params: ReadP): SCollection[Example] = {
+    val job = Job.getInstance()
+    GcsConnectorUtil.setInputPaths(sc, job, path)
+    job.setInputFormatClass(classOf[ExampleParquetInputFormat])
+    job.getConfiguration.setClass("key.class", classOf[Void], classOf[Void])
+    job.getConfiguration.setClass("value.class", classOf[Example], classOf[Example])
+
+    ParquetInputFormat.setReadSupportClass(job, classOf[ExampleReadSupport])
+    if (params.projection != null) {
+      ExampleParquetInputFormat.setFields(job, params.projection.asJava)
+    }
+    if (params.predicate != null) {
+      ParquetInputFormat.setFilterPredicate(job.getConfiguration, params.predicate)
+    }
+
+    val source = HadoopFormatIO
+      .read[JBoolean, Example]()
+      .withKeyTranslation(new SimpleFunction[Void, JBoolean]() {
+        override def apply(input: Void): JBoolean = true // workaround for NPE
```

Comment:
this means that this KeyTranslation is always called?

Reply:
`workaround for NPE` because Hadoop file IO always emits `<K, V>` and in this case null keys causes coder NPE.

Reply:
i would add this as a comment for the workaround NPE 😄